robotools/evotools/__init__.py

```diff
@@ -94,25 +94,48 @@ def _prepate_aspirate_dispense_parameters(
     # optional parameters
     if not isinstance(liquid_class, str) or ';' in liquid_class:
         raise ValueError(f'Invalid liquid_class: {liquid_class}')
+    
+    def _int_to_tip(tip_int: int):
+        """Asserts a Tecan Tip class to an int between 1 and 8."""
+        if not 1 <= tip_int <= 8:
+            raise ValueError(f"Tip is {tip} with type {type(tip)}, but should be an int between 1 and 8 for _int_to_tip conversion.")
+        if tip_int == 1:
+            return Tip.T1
+        elif tip_int == 2:
+            return Tip.T2
+        elif tip_int == 3:
+            return Tip.T3
+        elif tip_int == 4:
+            return Tip.T4
+        elif tip_int == 5:
+            return Tip.T5
+        elif tip_int == 6:
+            return Tip.T6
+        elif tip_int == 7:
+            return Tip.T7
+        elif tip_int == 8:
+            return Tip.T8
+
+
     if isinstance(tip, int) and not isinstance(tip, Tip):
-        if tip == 1:
-            tip = Tip.T1
-        elif tip == 2:
-            tip = Tip.T2
-        elif tip == 3:
-            tip = Tip.T3
-        elif tip == 4:
-            tip = Tip.T4
-        elif tip == 5:
-            tip = Tip.T5
-        elif tip == 6:
-            tip = Tip.T6
-        elif tip == 7:
-            tip = Tip.T7
-        elif tip == 8:
-            tip = Tip.T8
-    if not isinstance(tip, Tip):
-        raise ValueError(f'Invalid tip: {tip}')
+        # User-specified integers from 1-8 need to be converted to Tecan logic
+        tip = _int_to_tip(tip)
+
+    if isinstance(tip, collections.Iterable):
```

Comment:
```suggestion
    if isinstance(tip, collections.abc.Iterable):
```
Quoting [from stackoverflow](https://stackoverflow.com/a/4668647/4473230):
> Note: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working.